plugins/package-managers/spdx/src/main/kotlin/SpdxDocumentFile.kt

```diff
@@ -336,26 +336,36 @@ class SpdxDocumentFile(
     /**
      * Return the dependencies of the package with the given [pkgId] defined in [doc] of the
      * [SpdxRelationship.Type.DEPENDENCY_OF] type. Identified dependencies are mapped to ORT [Package]s and then
-     * added to [packages].
+     * added to [packages]. The [parents] set contains the IDs of the package that have already been encountered; it
+     * is used to detect circular dependencies.
      */
     private fun getDependencies(
         pkgId: String,
         doc: SpdxResolvedDocument,
-        packages: MutableSet<Package>
-    ): Set<PackageReference> =
-        getDependencies(pkgId, doc, packages, SpdxRelationship.Type.DEPENDENCY_OF) { target ->
+        packages: MutableSet<Package>,
+        parents: MutableSet<String>
```

Comment:
Could `parents` this have a default value?

Reply:
How about renaming this to `ancestorPackageIds` or just `ancestorIds` ?

Reply:
Renamed to `ancestorIds`. I decided against a default value to make sure that no invocation is missed.